protocol/rest/rest_server/go_restful_server.go

```diff
@@ -0,0 +1,310 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package rest_server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+)
+
+import (
+	"github.com/emicklei/go-restful/v3"
+	perrors "github.com/pkg/errors"
+)
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/protocol"
+	"github.com/apache/dubbo-go/protocol/invocation"
+	"github.com/apache/dubbo-go/protocol/rest/rest_interface"
+)
+
+func init() {
+	extension.SetRestServer(constant.DEFAULT_REST_SERVER, GetNewGoRestfulServer)
+}
+
+type GoRestfulServer struct {
+	srv       *http.Server
+	container *restful.Container
+}
+
+func NewGoRestfulServer() *GoRestfulServer {
+	return &GoRestfulServer{}
+}
+
+func (grs *GoRestfulServer) Start(url common.URL) {
+	grs.container = restful.NewContainer()
+	grs.srv = &http.Server{
+		Handler: grs.container,
+	}
+	ln, err := net.Listen("tcp", url.Location)
+	if err != nil {
+		panic(perrors.New(fmt.Sprintf("Restful Server start error:%v", err)))
+	}
+
+	go func() {
+		err := grs.srv.Serve(ln)
+		if err != nil && err != http.ErrServerClosed {
+			logger.Errorf("[Go Restful] http.server.Serve(addr{%s}) = err{%+v}", url.Location, err)
+		}
+	}()
+}
+
+func (grs *GoRestfulServer) Deploy(invoker protocol.Invoker, restMethodConfig map[string]*rest_interface.RestMethodConfig) {
+	svc := common.ServiceMap.GetService(invoker.GetUrl().Protocol, strings.TrimPrefix(invoker.GetUrl().Path, "/"))
+	for methodName, config := range restMethodConfig {
+		// get method
+		method := svc.Method()[methodName]
+		argsTypes := method.ArgsType()
+		replyType := method.ReplyType()
+		ws := new(restful.WebService)
+		ws.Path(config.Path).
+			Produces(strings.Split(config.Produces, ",")...).
+			Consumes(strings.Split(config.Consumes, ",")...).
+			Route(ws.Method(config.MethodType).To(getFunc(methodName, invoker, argsTypes, replyType, config)))
+		grs.container.Add(ws)
+	}
+
+}
+
+func getFunc(methodName string, invoker protocol.Invoker, argsTypes []reflect.Type, replyType reflect.Type, config *rest_interface.RestMethodConfig) func(req *restful.Request, resp *restful.Response) {
```

Comment:
this func declaration is so long. pls change it as follows:

func getFunc(methodName string, 
                        invoker protocol.Invoker,

Reply:
fix